validators.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const {
+  ArrayIsArray,
+} = require('./primordials');
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE
+  }
+} = require('./errors');
+
+function validateArray(value, name) {
+  if (!ArrayIsArray(value)) {
+    throw new ERR_INVALID_ARG_TYPE(name, 'Array', value);
+  }
+}
+
+/**
+ * @param {unknown} value
+ * @param {string} name
+ * @param {{
+ *   allowArray?: boolean,
+ *   allowFunction?: boolean,
+ *   nullable?: boolean
+ * }} [options]
+ */
+function validateObject(value, name, options) {
+  const useDefaultOptions = options == null;
+  const allowArray = useDefaultOptions ? false : options.allowArray;
+  const allowFunction = useDefaultOptions ? false : options.allowFunction;
+  const nullable = useDefaultOptions ? false : options.nullable;
```

Comment:
any reason not to use a module-level defaultOptions object, so it can be more like `(options ?? defaultOptions).allowArray`?

Reply:
Just that this code is vendored from Node.js:

https://github.com/nodejs/node/blob/master/lib/internal/validators.js#L153